packages/auth-construct/src/construct.ts

```diff
@@ -751,15 +751,15 @@ export class AmplifyAuth
       result.providersList.push('SAML');
     }
 
-    // UserPool Domain
-    if (this.domainPrefix && result.providersList.length > 0) {
+    // Auto generate Cognito Domain if external providers are configured
+    if (result.providersList.length > 0) {
```

Comment:
We should probably use [this](https://github.com/aws-amplify/amplify-backend/blob/ab739ea1ce7f0e297a28020976e51f8b1a2138e9/packages/backend/src/backend_id_scoped_stable_backend_identifiers.ts#L24-L26) recently added hash method. It's injected into the `generateContainerEntry` method in the `backend-auth` factory. I think the construct should continue to have an optional prop for the domainPrefix that can be computed by the factory and passed in.

Reply:
Good point, updated.